src/contracts/core/AllocationManager.sol

```diff
@@ -0,0 +1,731 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.27;
+
+import "@openzeppelin-upgrades/contracts/proxy/utils/Initializable.sol";
+import "@openzeppelin-upgrades/contracts/access/OwnableUpgradeable.sol";
+import "@openzeppelin-upgrades/contracts/security/ReentrancyGuardUpgradeable.sol";
+
+import "../permissions/Pausable.sol";
+import "../libraries/EIP1271SignatureUtils.sol";
+import "../libraries/SlashingLib.sol";
+import "./AllocationManagerStorage.sol";
+
+contract AllocationManager is
+    Initializable,
+    OwnableUpgradeable,
+    Pausable,
+    AllocationManagerStorage,
+    ReentrancyGuardUpgradeable
+{
+    using Snapshots for Snapshots.History;
+
+    /// @dev BIPS factor for slashable bips
+    uint256 internal constant BIPS_FACTOR = 10_000;
+
+    /// @dev Maximum number of pending updates that can be queued for allocations/deallocations
+    /// Note this max applies for a single (operator, strategy, operatorSet) tuple.
+    uint256 public constant MAX_PENDING_UPDATES = 1;
+
+    /// @dev Returns the chain ID from the time the contract was deployed.
+    uint256 internal immutable ORIGINAL_CHAIN_ID;
```

Comment:
Can we move all our constants/immutables to the same place? We have a handful sitting in the storage contract - let's keep them all there.

Reply:
Fixed in #786